sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonParser.scala

```diff
@@ -114,6 +120,18 @@ class JacksonParser(
     }
   }
 
+  private def failedRecord(record: String): Seq[InternalRow] = {
+    if (options.dropMalformed) {
+      if (!isWarningPrintedForMalformedRecord) {
+        logWarning(dropmalformedModeWarningMessage(record))
+        isWarningPrintedForMalformedRecord = true
+      }
+      Nil
+    } else {
+      throw new NotAllowedNullException(s"Null not allowed: $record")
```

Comment:
Please explain why null is not allowed in the error message. Is it possible to know which field is not nullable? I am not sure if just printing the record is really helpful at here (considering users may often have long JSON records).